clientconn.go

```diff
@@ -248,16 +248,26 @@ func Dial(target string, opts ...DialOption) (*ClientConn, error) {
 			return nil, errNoAddr
 		}
 	}
+
+	errC := make(chan error, len(addrs))
+	for _, addr := range addrs {
+		go func(a Address) {
+			errC <- cc.newAddrConn(a, false)
+		}(addr)
+	}
 	waitC := make(chan error, 1)
 	go func() {
-		for _, a := range addrs {
-			if err := cc.newAddrConn(a, false); err != nil {
-				waitC <- err
+		defer close(waitC)
+		var errs []error
+		for range addrs {
+			errs = append(errs, <-errC)
```

Comment:
surely this can be written as the less surprising
```
if err := <-errC; err != nil {
  errs = append(errs, err)
} else {
  return
}
```
or 
```
err := <-errC
if err == nil {
  return
}
errs = append(errs, err)
```